tools/db_bench_tool.cc

```diff
@@ -789,6 +789,27 @@ DEFINE_int32(min_level_to_compress, -1, "If non-negative, compression starts"
              " not compressed. Otherwise, apply compression_type to "
              "all levels.");
 
+typedef enum rocksdb::BlockBasedTableOptions::DataBlockIndexType
+    DataBlockIndexType;
+
+static DataBlockIndexType StringToDataBlockIndexType(const char* ctype) {
+  assert(ctype);
+
+  if (!strcasecmp(ctype, "binary"))
+    return rocksdb::BlockBasedTableOptions::kDataBlockBinarySearch;
+  else if (!strcasecmp(ctype, "binary_and_hash"))
+    return rocksdb::BlockBasedTableOptions::kDataBlockBinaryAndHash;
+
+  fprintf(stdout, "Cannot parse data block index type '%s'\n", ctype);
+
+  // return default value
+  return rocksdb::BlockBasedTableOptions::kDataBlockBinarySearch;
+}
+
+DEFINE_string(data_block_index_type, "binary", "Index type for data blocks");
+static DataBlockIndexType FLAGS_data_block_index_type_e =
+    rocksdb::BlockBasedTableOptions::kDataBlockBinarySearch;
```

Comment:
Almost all block based table options seem to be defined between lines 415 - 467. Could you move these two new params also to be close to that area of code?